docs/designers-developers/developers/internationalization.md

```diff
@@ -1,47 +1,157 @@
+
+```php
+<?php
+	function myguten_set_script_translations() {
+		wp_set_script_translations( 'myguten-script', 'myguten' );
+	}
+	add_action( 'init', 'myguten_set_script_translations' );
+```
+
+This is all you need to make your plugin JavaScript code translatable.
+
+When you set script translations for a handle WordPress will automatically figure out if a translations file exists on translate.wordpress.org, and if so ensure that it's loaded into `wp.i18n` before your script runs.  With translate.wordpress.org, plugin authors also do not need to worry about setting up their own infrastructure for translations and can rely on a global community with dozens of active locales. Read more about [WordPress Translations](https://make.wordpress.org/meta/handbook/documentation/translations/).
+
+## Provide Your Own Translations
+
+You can create and ship your own translations with your plugin, if you have sufficient knowledge of the language(s) you can ensure the translations are available.
+
+### Create Translation File
+
+The translation files must be in the JED 1.x JSON format.
+
+To create a JED translation file, first you need to extract the strings from the text. Typically, the language files all live in a directory called `languages` in your plugin.  Using [WP-CLI](https://wp-cli.org/), you create a `.pot` file using the following command from within your plugin directory:
+
+```
+mkdir languages
+wp i18n make-pot ./ languages/myguten.pot
+```
+
+This will create the file `myguten.pot` which contains all the translatable strings from your project. A single entry in this POT file might look like this:
+
+```
+#: block.js:6
+msgid "Simple Block"
+msgstr ""
+```
+
+Here, `msgid` is the string to be translated, and `msgstr` is the actual translation. In the POT file, `msgstr` will always be empty.
+
+This POT file can then be used as the template for new translations. You should copy the file using the language code you are going to translate, this example will use the Esperanto (eo) language:
+
+```
+cp myguten.pot myguten-eo.po
+```
+
+For this simple example, you can simply edit the `.po` file in your editor and add the translation to all the `msgstr` sets. For a larger, more complex set of translation, the [Glotpress](https://glotpress.blog/) and [poedit](https://poedit.net/) tools exist to help.
+
+```
+#: block.js:6
+msgid "Simple Block"
+msgstr "Simpla bloko"
+```
+
+
+The last step to create the translation file is to convert the `myguten-eo.po` to the JSON format needed. For this, you use the [po2json utility](https://github.com/mikeedwards/po2json) which you can install using npm. It might be easiest to install globally using: `npm install -g po2json`. Once installed, use the following command to convert to JED format:
+
+```
+po2json myguten-eo.po myguten-eo.json -f jed
```

Comment:
I'm running into this error when executing this command:

![po2json-plural-form-errors](https://user-images.githubusercontent.com/583546/53976985-6154c500-4108-11e9-960b-a94dd2f3dc16.png)

Reply:
We could also include the output of this command, so the user knows what to expect (and isn't stuck if the command fails for some reason).

Reply:
Added the JSON file which should be created.